### PR TITLE
Handle broker nonce in common core

### PR DESCRIPTION
**Pull request by jasoncoolmax**

### CODE REVIEW

File: IdentityCore/src/requests/MSIDBrokerTokenRequest.m [hunk 1]
```diff
@@ -174,7 +177,8 @@ - (NSDictionary *)defaultResumeDictionaryContents
       @"redirect_uri"     : self.requestParameters.redirectUri,
       @"correlation_id"   : self.requestParameters.correlationId.UUIDString,
 #if TARGET_OS_IPHONE
-      @"keychain_group"   : self.requestParameters.keychainAccessGroup ?: MSIDKeychainTokenCache.defaultKeychainGroup
+      @"keychain_group"   : self.requestParameters.keychainAccessGroup ?: MSIDKeychainTokenCache.defaultKeychainGroup,
+      @"broker_nonce"     : self.brokerNonce
```
Comment (oldalton) [inline IdentityCore/src/requests/MSIDBrokerTokenRequest.m]: nit: can broker nonce be nil?

Comment (jasoncoolmax) [reply IdentityCore/src/requests/MSIDBrokerTokenRequest.m]: To make sure it won't be nil, I added a getter to always init the nonce if it is not set yet.

File: IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m [hunk 2]
```diff
@@ -219,12 +219,12 @@ - (void)saveToPasteBoard:(NSURL *)url
...
 + (BOOL)completeAcquireToken:(nullable NSURL *)resultURL
-           sourceApplication:(nonnull NSString *)sourceApplication
+           sourceApplication:(nullable NSString *)sourceApplication
        brokerResponseHandler:(nonnull MSIDBrokerResponseHandler *)responseHandler
 {
-    BOOL isBrokerResponse = [self isResponseFromBroker:sourceApplication];
-
-    if (!isBrokerResponse)
+    // sourceApplication could be nil, we want to return early if we know for sure response is not from broker
+    BOOL isNotFromBroker = sourceApplication && ![self isResponseFromBroker:sourceApplication];
```
Comment (unpluggedk) [inline IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: nit: let's do isFromBroker and do a !check on the if. just to get into habit of not creating something contradicting in first place.

Also, is this the right condition?  (i.e./ this means
isFromBroker = !sourceApplication || [self isResponseFromBroker] .
I am not sure if we can conclude isFromBroker is there is no sourceApplication.

File: IdentityCore/src/requests/sdk/MSIDBrokerResponseHandler.m [hunk 3]
```diff
@@ -177,6 +182,13 @@ - (NSDictionary *)verifyResumeStateDicrionary:(NSURL *)response error:(NSError *
         MSIDFillAndLogError(error, MSIDErrorBrokerBadResumeStateFound, @"Resume state is missing the redirect uri!", correlationId);
         return nil;
     }
+    
+    NSString *brokerNonce = [resumeDictionary objectForKey:@"broker_nonce"];
+    if (!brokerNonce)
+    {
+        MSIDFillAndLogError(error, MSIDErrorBrokerBadResumeStateFound, @"Resume state is missing the broker nonce!", correlationId);
```
Comment (unpluggedk) [inline IdentityCore/src/requests/sdk/MSIDBrokerResponseHandler.m]: isn't it broker nonce is missing in the resume state?

**Comment by jasoncoolmax** [comment]
> nit: we might want to use another word instead of "nonce".

discussed offline. "nonce" is a generic term, so we agree to stick with the name here.

File: IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m (hunk 2 above)
Comment (jasoncoolmax) [reply IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m]: agree, addressed :)

**Pull request merged**